### PR TITLE
[Packager] Backslashes should be escaped correctly under Windows box

### PR DESCRIPTION
**Pull request by compulim**
Inside the packaged `index.ios.bundle`, some lines need to be fixed:

##### Definition of the main module
From
`__d('myapp\index.ios.js',["react-native\\Libraries\\react-native\\react-native.js"]`
to
`__d('myapp\\index.ios.js',["react-native\\Libraries\\react-native\\react-native.js"]`

##### References from main module
From
`var React=require('react-native\Libraries\react-native\react-native.js');`
to
`var React=require('react-native\\Libraries\\react-native\\react-native.js');`

##### Entrypoint at the end of the file
From
`;require("myapp\index.ios.js");`
to
`;require("myapp\\index.ios.js");`

Instead of turning backslashes into slashes, I would prefer keeping it the backslashes and escape correctly. This would minimize code changes and reduce risks.

### CODE REVIEW

**Comment by compulim** [comment]
Looks like the `[RCTAllocationTests testModulesAreDeallocated]` test is not very stable. I see it also failing on other PRs. My changes are only on packager part.

File: packager/react-packager/src/DependencyResolver/index.js [hunk 1]
```diff
@@ -186,7 +186,7 @@ HasteDependencyResolver.prototype.getDebugInfo = function() {
 function defineModuleCode({moduleName, code, deps}) {
   return [
     `__d(`,
-    `'${moduleName}',`,
+    `'${moduleName}',`.replace(replacePatterns.BACKSLASHES_RE, '\\\\'),
```
Comment (ide) [inline packager/react-packager/src/DependencyResolver/index.js]: can you put the replace() call inside of the `${}` so it explicitly escapes just the module name?

**Comment by ide** [comment]
It might be better to use JSON.stringify which would handle backslashes and almost all other cases we haven't thought of yet.

**Comment by compulim** [comment]
@ide Please take a look. It's a great idea to use `JSON.stringify` to escape strings.

**Comment by ide** [comment]
Looks good to me at a high level. I like what you did with start/endQuot (we could even make those non-capturing groups since they're never used... will leave that to @amasad to decide). Could you please squash your commits (in general we prefer one Git commit per PR)?

**Comment by compulim** [comment]
@ide I have squashed my commits.

**Comment by compulim** [comment]
@amasad Is there anything I could do to smoothen this PR?

**Comment by amasad** [comment]
A quick look, this looks good. Longterm I want to find a way to do use the AST to do this kind of work without compromising on perf.

We're actually in the process of moving the packager to it's own github repo and we'll be able to accept contributions like this much faster. I'm hoping we can release it some time this week. Thanks!

**Comment by compulim** [comment]
@amasad Please let me know when the new repo for packager is ready. Thanks!

**Comment by mkonicek** [comment]
@amasad Any updates?

**Comment by facebook-github-bot-4** [comment]
@martinbigio would you mind taking a look at this pull request? It's been a while since the last commit was reviewed.

**Comment by martinbigio** [comment]
The new bots are working woohoo!

I'm fine with this change but there has been lots of changes since the base rev of this PR so not sure if it's still necessary. Please rebase and re-test if it is but please wait a couple of days as @davidaurelio is about to land a big refactor.

**Comment by mkonicek** [comment]
@compulim Can you rebase please?

**Comment by compulim** [comment]
Thanks, working on it now.

**Comment by compulim** [comment]
Looks like the changes has already been incorporated in 06b5bda. Thanks everyone..

**Pull request closed without merging**